From: Ingo Tuchscherer <ingo.tuchscherer@linux.vnet.ibm.com>
Subject: s390/zcrypt: Introduce CEX6 toleration
Patch-mainline: v4.10-rc1
Git-commit: b3e8652bcbfa04807e44708d4d0c8cdad39c9215
References: bsc#1020686, FATE#321530, LTC#146546

Summary:     zcrypt: toleration of new crypto adapter hardware with type 12
Description: This feature implements toleration of ap bus devices with
             device type 12 and treat them like CEX5S crypto adapters.

Upstream-Description:

             s390/zcrypt: Introduce CEX6 toleration

             Signed-off-by: Harald Freudenberger <freude@linux.vnet.ibm.com>
             Signed-off-by: Martin Schwidefsky <schwidefsky@de.ibm.com>


Signed-off-by: Ingo Tuchscherer <ingo.tuchscherer@linux.vnet.ibm.com>
Acked-by: John Jolly <jjolly@suse.com>
---
 drivers/s390/crypto/ap_bus.c |    2 ++
 drivers/s390/crypto/ap_bus.h |    1 +
 2 files changed, 3 insertions(+)

--- a/drivers/s390/crypto/ap_bus.c
+++ b/drivers/s390/crypto/ap_bus.c
@@ -1651,6 +1651,8 @@ static void ap_scan_bus(struct work_stru
 		ap_dev->queue_depth = queue_depth;
 		ap_dev->raw_hwtype = device_type;
 		ap_dev->device_type = device_type;
+		if (device_type == AP_DEVICE_TYPE_CEX6)
+			ap_dev->device_type = AP_DEVICE_TYPE_CEX5;
 		ap_dev->functions = device_functions;
 		spin_lock_init(&ap_dev->lock);
 		INIT_LIST_HEAD(&ap_dev->pendingq);
--- a/drivers/s390/crypto/ap_bus.h
+++ b/drivers/s390/crypto/ap_bus.h
@@ -105,6 +105,7 @@ static inline int ap_test_bit(unsigned i
 #define AP_DEVICE_TYPE_CEX3C	9
 #define AP_DEVICE_TYPE_CEX4	10
 #define AP_DEVICE_TYPE_CEX5	11
+#define AP_DEVICE_TYPE_CEX6	12
 
 /*
  * Known function facilities
